cmake/FindRust.cmake

```diff
@@ -166,9 +166,19 @@ if (_RESOLVE_RUSTUP_TOOLCHAINS)
             NO_DEFAULT_PATH)
 else()
     find_program(Rust_COMPILER_CACHED rustc)
+    if (Rust_COMPILER_CACHED)
+        get_filename_component(_RUST_TOOLCHAIN_PATH ${Rust_COMPILER_CACHED} DIRECTORY)
+        get_filename_component(_RUST_TOOLCHAIN_PATH ${_RUST_TOOLCHAIN_PATH} DIRECTORY)
+    endif()
+endif()
 
-    get_filename_component(_RUST_TOOLCHAIN_PATH ${Rust_COMPILER_CACHED} DIRECTORY)
-    get_filename_component(_RUST_TOOLCHAIN_PATH ${_RUST_TOOLCHAIN_PATH} DIRECTORY)
+if (NOT Rust_COMPILER_CACHED)
+    message(
+        WARNING "The rustc executable was not found. "
+        "Rust not installed or ~/.cargo/bin not added to path? "
```

Comment:
> ~/.cargo/bin not added to path?

That should be trivial to check automatically, and print even more descriptive message.

Reply:
Yeah, I am not sure what the best actual check would be for doing this.
I did this PR as a more minimal improvement, but there is likely a better check which can be done alongside a refactor to make it more clear.
You have any suggestions? Because the .cargo/bin could probably be placed in different paths, so it needs to support just searching for the rustc executable in path I guess.

Reply:
@ratijas You think this is a fine first step, or do you want me to do any larger changes?

Reply:
Ship it.

Even if wording is not 100% technically correct and ommits some non-standard setups, it still nudges user in the right direction.